File: Hot 100/78. 子集.cpp
class Solution {
private:
    vector<vector<int>> ans;
    bool vis[20];
public:
    void dfs(int ind,vector<int>& nums,vector<int> tmp){
        if(ind==nums.size()){
            ans.push_back(tmp);
            return;
        }
        tmp.push_back(nums[ind]);
        dfs(ind+1,nums,tmp);
        tmp.pop_back();
        dfs(ind+1,nums,tmp);
        
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<int> tmp;
        dfs(0,nums,tmp);
        return ans;
    }
};

File: Hot 100/78. 子集.go
func subsets(nums []int) [][]int {
	ans := [][]int{}
	var dfs func(int, []int)
	dfs = func(ind int, tmp []int) {
		if ind == len(nums) {
			ans = append(ans, append([]int(nil), tmp...))
			return
		}
		tmp = append(tmp, nums[ind])
		dfs(ind+1, tmp)
		tmp = tmp[:len(tmp)-1]
		dfs(ind+1, tmp)
	}
	dfs(0, []int{})
	return ans
}